test/cpp/jit/test_gpu.cpp

```diff
@@ -9855,6 +9855,83 @@ TEST(NVFuserTest, FusionLoopUnswitch_CUDA) {
       &fusion, outputs, aten_inputs, {aten_output}, __LINE__, __FILE__);
 }
 
+TEST(NVFuserTest, FusionManualMultiKernel_CUDA) {
+  constexpr int bid_x = 80;
+  constexpr int tid_x = 4096;
+
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // Set up your input tensor views
+  TensorView* tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  TensorView* tv1 = sum(tv0, {0});
+
+  TensorView* tv2 = add(tv1, tv0); // implicit bcast
+
+  TensorView* tv3 = sum(tv2, {1});
+
+  fusion.addOutput(tv3);
+
+  const auto options =
+      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  at::Tensor aten_input = at::randn({bid_x, tid_x}, options);
+  auto aten_output =
+      aten_input.to(at::kDouble).sum({0}).add(aten_input).sum({1});
+
+  // Setup and run first fusion
+
+  Fusion fusion0;
+  auto clone0 = Fusion::copy(&fusion, &fusion0);
+
+  fusion0.removeOutput(clone0.clone(tv3));
+  fusion0.addOutput(clone0.clone(tv1));
+
```

Comment:
I can see why we may need the mapping, but I don't like exposing the cloning details like this. Not only that the IR cloning is supposed to be an internal helper, but the interface is non-intuitive (at least to me):

- Having both a Fusion::copy() and the usual copy/move operations 
- using `clone()` to extract the mapping is misleading (no cloning actually happens) and depends on knowledge of how cloning works (creating a hidden tight coupling)

Reply:
We can invent better interfaces (like the one proposed by @naoyam), but ideally we should avoid exposing the cloning machinery altogether: when we need to map node we should be able to use lookups using the node identities in the new fusion to locate the corresponding nodes.

(In KIR there's a new `kir::Val::id()` exactly for thing purpose, although I haven't got that to the final conclusion yet. But for Fusion IR, the names should serve this purpose nicely)

Reply:
I agree with @tlemo, but I also didn't think this would be a big deal since manual cloning shouldn't be something a user would routinely play with. So, as a step toward automatic kernel splitting, perhaps this would be just fine?

Reply:
The problem is that once you leak internal details (and create hidden coupling) is _much_ harder to "fix" it later compared to avoiding the problem in the first place.

Reply:
That's indeed true.

Reply:
This conversation seems a bit vague to me, can you propose an API that would address your concerns @tlemo. To me there is a very strong coupling between the fusions and trying to avoid that basic truth by trying to recompute maps with an id seems more misleading to me. I'm fine with being more explicit, though uncertain what that would look like beyond Naoya's proposal.

Reply:
What is the problem with looking up relevant nodes using their names?

Reply:
It seems just as internal of a detail as having an api that returns a map. Except one provides a map that has to be generated to produce the clone and the other you reconstruct that map through a relatively arbitrary side effect of the clone which is names remain consistent.

Reply:
What I mean is a general name -> node lookup, independent of cloning:

```cpp
  fusion0.removeOutput(fusion0.lookupNode(tv3->name()));
  fusion0.addOutput(fusion0.lookupNode(tv1->name()));
```

This `lookupNode()` method would be part of the Fusion interface (internally we'd just have a map with names -> node ptrs)